packages/react-native/src/breadcrumbs/events/AppStateBreadcrumbSubscriber.ts

```diff
@@ -0,0 +1,63 @@
+import {
+    BreadcrumbLogLevel,
+    BreadcrumbType,
+    type BacktraceBreadcrumbs,
+    type BreadcrumbsEventSubscriber,
+} from '@backtrace/sdk-core';
+import { AppState, Platform, type NativeEventSubscription } from 'react-native';
+
+export class AppStateBreadcrumbSubscriber implements BreadcrumbsEventSubscriber {
+    private _nativeEventSubscriptions: NativeEventSubscription[] = [];
+    public start(backtraceBreadcrumbs: BacktraceBreadcrumbs): void {
+        if ((backtraceBreadcrumbs.breadcrumbsType & BreadcrumbType.System) === BreadcrumbType.System) {
```

Comment:
why should the subscriber care if the level is `System` or not?

Reply:
because we don't even need to set the connection between native and managed layer. Thats why I wrapped them in the if statement

Reply:
makes sense. Thanks